layers/newrelic/README.md

```diff
@@ -0,0 +1,85 @@
+# Bref New Relic Extension
+
+This is a Dockerfile to help add New Relic's PHP Agent to the Bref runtime so monitoring your applications on Lambda is possible. 
+
+## Install and configure
+
+```cli
+composer require bref/extra-php-extensions
+```
+
+```yaml
+# serverless.yml
+service: app
+
+provider:
+    name: aws
+    region: us-east-1
+    runtime: provided
+
+plugins:
+    - ./vendor/bref/bref
+    - ./vendor/bref/extra-php-extensions # <----- Add the extra Serverless plugin
+
+functions:
+    console:
+        handler: bin/console
+        layers:
+            - ${bref:layer.php-74}
+            - ${bref:extra.newrelic-php-74}
+            - ${bref:layer.console}
+```
+
+### Extra configuration files
+
+The configuration is only set to call newrelic.so to load the PHP Agent. In order to make this work you must configure the agent settings in your application and pass a name for the monitored application, a license key for your account and the IP/address of the New Relic PHP daemon listening in ECS or on a machine. 
+
+You can if you wish set these settings as Environment Variables and set the values in the Lambda Environment Variables to control these settings at runtime. If you use Environment variables it's recommended to use some key management service like KMS.
+
+You should configure your application in `php/conf.d/newrelic-settings.ini` within your Bref Application. This will mean it's controlled by your application code. 
+
+```yaml
+    NEW_RELIC_APP_NAME: "BrefApp"
+    NEW_RELIC_LICENSE_KEY: "ceaxxxxxxxxxxxxxxxxxxxxceafe3"
+    NEW_RELIC_DAEMON_ADDRESS: "xx.xx.xxx.xxx:31339"
+    NEW_RELIC_LOG_FILE: "/proc/self/fd/2"
+    NEW_RELIC_LOG_LEVEL: "info"
+    NEW_RELIC_DAEMON_APP_CONNECT_TIMEOUT: "5s"
+    NEW_RELIC_DAEMON_START_TIMEOUT: "5s"You may add extra 
+```   
+
+We would recommend pointing the logfile to `/proc/self/fd/2` which is STDERR and you will therefore get logging to Cloudwatch in the event of an issue. Log Level defaults to INFO but if you only want issues and not conversational logging you can raise the level to `warning` or `error`. Daemon address should be the IP of the location of your remotely started daemon (see below for guidance.)
```

Comment:
Are you sure about that?

I tried again (because I did already a year ago) and it doesn't seem possible to write to that file inside lambda:

```php
    $stderr = fopen('/proc/self/fd/2', 'ab');
    fwrite($stderr, 'This is a test log into stderr');
    fclose($stderr);
```

<img width="1151" alt="Screen 20201221 2ZFSpHLy@2x" src="https://user-images.githubusercontent.com/720328/102808539-be466b80-43c0-11eb-8c02-d652e877ae79.png">

Reply:
It's logged to Cloudwatch logs. So I can see what the PHP agent is trying to do using that. Here's an example. Without it if you have issues it'll be hard to see what the agent was trying to do. Lambda being read only this is the only approach I found that has worked.

<img width="1555" alt="Screenshot 2020-12-23 at 00 45 54" src="https://user-images.githubusercontent.com/28544519/102946357-4ce9e400-44b8-11eb-9a62-356e13f500e7.png">

Reply:
I'm not sure how it works for the daemon and not the lambda function, but if it works, it works! Thanks!